set route, TODO link css file

pilna opened the pull request:
basic REST api with a few html template files

add modele file

add correspondance extension name of language

add coloration page

add create snippet function

add: core feature, upload new bin and retrieve it

This commit includes the core feature : upload a new snippet via an HTML
form and get back an unique identifier to that snippet. A basic syntax
highlighting feature is implemented too.

There is no database yet, the snippets are stored in memory.

Co-authored-by: Pilna <pilna@outlook.fr>
Co-authored-by: wyx0-xyz <wyxo.bs09@gmail.com>

Closes #4
Related to #8

pilna committed:
basic REST api with a few html template files

Julien00859 commented on bin/__init__.py:
```diff
@@ -0,0 +1,4 @@
+from bottle import TEMPLATE_PATH
+
+root = __import__("pathlib").Path(__file__).resolve().parent
+TEMPLATE_PATH.insert(0, str(root/"view"))
```
Il manque le retour à la ligne à la fin du fichier. Aussi, je n'aime pas trop que le symbole `TEMPLATE_PATH` finisse dans le `__init__.py`, dans l'état si tu fais `import bin; dir(bin)` tu verras `root` (ce qui est souhaitable) et `TEMPLATE_PATH` ce qui ne l'est pas ? Après ça ne gène pas trop non plus donc on peut laisser ainis :)

Julien00859 commented on bin/assets/css/style.css:
```diff
@@ -0,0 +1,3 @@
+.red {
+    color: red;
+}
```
Il manque le retour à la ligne à la fin du fichier

Julien00859 commented on bin/bin.py:
```diff
@@ -1,5 +1,20 @@
+def publish_new_snippet():
+    return 'publish_new_snippet'
+
+@route('/<ids>', method='GET')
```
J'aime pas la variable `ids` le s sous-entend qu'il pourrait y en avoir plusieurs alors que ce n'est pas le cas. J'ai vu passer du code où la variable s'appelait `snippet`, je trouve ça mieux.

Julien00859 commented on bin/view/index.html:
```diff
@@ -0,0 +1,9 @@
+    </textarea>
+    <input type="submit" value="valider" />
+</form>
+% include('layouts/footer.html')
```
Retour à la ligne

Julien00859 commented on bin/view/layouts/footer.html:
```diff
@@ -0,0 +1,2 @@
+    </body>
+</html>
```
Retour à la ligne

Julien00859 commented on bin/view/layouts/header.html:
```diff
@@ -0,0 +1,11 @@
+        <meta charset=utf-8>
+        <title>bin</title>
+        <link rel="stylesheet" href="bin/assets/css/style.css">
+        <script type="text/javascript" src="#"></script>
```
Il y a un mélange entre des attributs html quotés et pas quotés, il faut choisir une convention et s'y tenir. Ma convention perso c'est de ne jamais quote sauf pour: `src`, `href`, `value` et `class`.

Julien00859 reviewed (changes requested):
Quelques modifs minimales, surtout des retours à la ligne en fin de fichiers qui manquent sinon bon boulot ! Merci :)

Julien00859 commented on bin/bin.py:
```diff
@@ -1,6 +1,11 @@
 
+@route('/assets/<dir>/<filepath:path>')
+def styles(dir=None, filepath=None):
+    return static_file(path.join(dir, filepath), root=root.joinpath('assets'))
```
No need for the `<dir>` placeholder, also why is the function named `styles` ?

    @route('/assets/<filepath:path>')
    def assets(filepath):
        return static_files(filepath, root=root/'assets')

pilna committed:
add modele file

pilna committed:
add correspondance extension name of language

pilna committed:
add coloration page

pilna committed:
add create snippet function

NoeMastrorillo reviewed (dismissed):
Modifier le footer et le header pour que ça colle avec la structure décidée en conversation :
footer.html : 
```
<footer>
    <!-- ... -->
</footer>
```
header.html : 
```
<head>
    <!-- ... -->
</head>
```

Julien00859 commented on bin/bin.py:
```diff
@@ -1,5 +1,33 @@
+@route('/<snippetId>.<ext>', method='GET')
+def display_with_coloration(snippetId, ext=None):
+    code = get_code_by_snippetId(snippetId)
+    language = languageOfExtension[ext] if ext in languageOfExtension else ""
```
`languageOfExtension.get(ext, "")`

aussi j'aime pas le camelCase

Julien00859 commented on bin/bin.py:
```diff
@@ -1,5 +1,33 @@
+
+@route('/raw/<snippetId>', method='GET')
+@route('/raw/<snippetId>.<ext>', method='GET')
+def display_text(snippetId, ext=None):
```
j'aime pas le camelCase

Julien00859 commented on bin/models/code.py:
```diff
@@ -0,0 +1,16 @@
+def get_code_by_snippetId(snippetId):
```
snamel_Case les gars !

Julien00859 commented on bin/views/layouts/header.html:
```diff
@@ -0,0 +1,11 @@
+    <link rel=stylesheet href="//cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.3.2/build/styles/default.min.css">
+    <script src="//cdn.jsdelivr.net/gh/highlightjs/cdn-release@10.3.2/build/highlight.min.js"></script>
+    <script type=text/javascript src="#"></script>
+    <script>hljs.initHighlightingOnLoad();</script>
```
hightlightjs n'est pas nécessaire pour la page new bin

Julien00859 committed:
add: core feature, upload new bin and retrieve it

This commit includes the core feature : upload a new snippet via an HTML
form and get back an unique identifier to that snippet. A basic syntax
highlighting feature is implemented too.

There is no database yet, the snippets are stored in memory.

Co-authored-by: Pilna <pilna@outlook.fr>
Co-authored-by: wyx0-xyz <wyxo.bs09@gmail.com>

Closes #4
Related to #8
